Shared/Samples/Trace utility network/TraceUtilityNetworkView.Views.swift

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension TraceUtilityNetworkView {
+    /// The buttons and picker shown to the user while setting points.
+    @ViewBuilder
+    private var controlsForSettingPoints: some View {
+        Group {
+            Picker("Add starting points & barriers", selection: pointType) {
+                ForEach([PointType.start, PointType.barrier], id: \.self) { type in
+                    Text(type.label).tag(type)
+                }
+            }
+            .padding()
+            .pickerStyle(.segmented)
+            Button("Trace") {
+                model.tracingActivity = .traceRunning
+            }
+            .disabled(model.pendingTraceParameters?.startingLocations.isEmpty ?? true)
+            .padding()
+            Button("Reset", role: .destructive) {
+                model.reset()
+            }
+            .padding()
+        }
+        .disabled(model.tracingActivity == .traceRunning)
+    }
+    
+    /// A binding to the type of the starting point being set.
+    ///
+    /// - Note: This should only be used when the user is setting starting points or barriers. If
+    /// this condition isn't present, gets will be inaccurate and sets will be ignored.
+    private var pointType: Binding<PointType> {
+        .init(
+            get: {
+                guard case .settingPoints(let pointType) = model.tracingActivity else {
+                    return .start
+                }
+                return pointType
+            },
+            set: {
+                guard case .settingPoints = model.tracingActivity else { return }
+                model.tracingActivity = .settingPoints(pointType: $0)
+            }
+        )
+    }
+    
+    /// Buttons for each the available terminals on the last added utility element.
+    @ViewBuilder
+    var terminalPickerButtons: some View {
+        ForEach(model.lastAddedElement?.assetType.terminalConfiguration?.terminals ?? []) { terminal in
+            Button(terminal.name) {
+                model.lastAddedElement?.terminal = terminal
+                model.updateUserHint(withMessage: "terminal: \(terminal.name)")
+            }
+        }
+    }
+    
+    /// The menu at the bottom of the screen that guides the user through running a trace.
+    var traceMenu: some View {
+        HStack(spacing: 5) {
```

Comment:
Something is odd about this, removing the `HStack` causes no UI to be drawn at all, even with `controlsForSettingPoints` being an `HStack` itself now. It might be something with `controlsForSettingPoints` needing to anchor itself to another view inside of the toolbar. I can revisit and spend more time on this if you like.

Reply:
This might be yet another SwiftUI bug (or feature). On iOS 15, `Group` works just fine.

I actually thought making both `Group` (i.e., `controlsForSettingPoints` and `traceMenu`) makes more sense. Because the `traceMenu` is in a `ToolbarItemGroup`, the layout should already be managed by the toolbar, thus we don't need to use an `HStack`.